### PR TITLE
Add some docs on the format and context of started/finished.json

### PR DESCRIPTION
**Pull request by MushuEE**
/area prow
/kind documentation
I wrote this document in response to #3412 which can probably be closed. Started and Finished JSON files have not been consistent and there has been some confusion around what they contain. Hopefully this document clears that up.

/cc @BenTheElder

### CODE REVIEW

**Comment by MushuEE** [comment]
Original Document: https://docs.google.com/document/d/1HuMOGJ_jkxP4aktMp1C7Bo-sENGmvbeywc44m3mhdUc/edit?usp=sharing

File: prow/started_and_finished.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Understanding Started.json and Finished.json
```
Comment (amwat) [inline prow/started_and_finished.md:1]: nit: something like metadata_artifacts.md would probably be more apt than just started and finished.
(eventually) we would also want other files documented here as well.

**Review by amwat — COMMENTED** [review]
/lgtm

File: prow/started_and_finished.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+{"timestamp":1595277241,
+"pull":"93264",
+"repos":{"kubernetes/kubernetes":"master:5feab0aa1e592ab413b461bc3ad08a6b74a427b4,93264:5dd9241d43f256984358354d1fec468f274f9ac4"},
+"metadata":
+   {"links":
+      {"resultstore":
+         {"url":"https://source.cloud.google.com/results/invocations/20688dbb-eb32-47e6-8a49-34734e714f81/targets/test"}},
+    "resultstore":"https://source.cloud.google.com/results/invocations/20688dbb-eb32-47e6-8a49-34734e714f81/targets/test"},
+      "repo-version":"30f64c5b1fc57a3beb1476f9beb29280166954d1",
+      "Pending":false}
```
Comment (petr-muller) [inline prow/started_and_finished.md:104]: Can we pretty print this?

File: prow/started_and_finished.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+|repo-version (dep)|*unset*|
+|job-version (dep)|*unset*|
+|revision (dep)|[SHA from Refs](https://github.com/kubernetes/test-infra/blob/4b5c7c99a851eb427f5c77bd0c8d11526f7b63c4/prow/pod-utils/downwardapi/jobspec.go#L163)|
+|metadata| *unset*|
+*Ex*
+```
+{"timestamp":1595279434,
+"passed":true,
+"result":"SUCCESS",
+"revision":"5dd9241d43f256984358354d1fec468f274f9ac4"}
```
Comment (petr-muller) [inline prow/started_and_finished.md:122]: ANd this too?

File: prow/started_and_finished.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+|version|If version is in metadata, set from metadata same as job-version|
+|result|'SUCCESS' or "FAILURE' depending on passed|
+|job-version (dep)|If not existing and not 'unknown'... from metadata, try 'job-version' then 'version'|
+|metadata|exact same as `repo-version`|
+|- repo-commit|Git rev-parse HEAD (for k8s)|
+|- repos|Same as started 'comes from --repo= args'|
+|- infra-commit|Git rev-parse HEAD (for test-infra)|
+|- repo|main repo for job|
+|- job-version| Same as job version from above|
+|- revision| Same as job-version|
```
Comment (petr-muller) [inline prow/started_and_finished.md:60]: The notation with `-` for sub-members looks a bit weird, maybe we should use jsonpath and also format the fields with monospace, e.g.`.metadata.repos` etc

File: prow/started_and_finished.md (hunk 1 above)
Comment (MushuEE) [reply prow/started_and_finished.md:1]: Done

File: prow/started_and_finished.md (hunk 2 above)
Comment (MushuEE) [reply prow/started_and_finished.md:104]: made pretty

File: prow/started_and_finished.md (hunk 4 above)
Comment (MushuEE) [reply prow/started_and_finished.md]: Like it, done

**Review by petr-muller — APPROVED** [review]
/retest

**Comment by MushuEE** [comment]
@amwat quick poke

File: prow/metadata_artifacts.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
+# Understanding Started.json and Finished.json
+
+### Context
+Prow uploads a host of artifacts throughout the life cycle of a job. Two of these artifacts that are present in each run are `started.json` and `finished.json` which contain a host of information pertaining to the job/run. These files have existed through the evolution of Kubernetes CI: from Jenkins -> Containerized Jenkins -> Bootstrap Containerized Jenkins -> Bootstrap Prow -> PodUtils. As of 2021, all jobs exist within either Bootstrap Prow or PodUtils. As the CI has evolved, so has `started/finished.json` and it's function.
+
+Related Issues:
+
+1. #3412: What is the origin and porous for the fields in these files?
+2. #111000: This isn't a source of truth and prow/pod/gcs are not in sync
```
Comment (amwat) [inline prow/metadata_artifacts.md:9]: is this the right issue number?

File: prow/metadata_artifacts.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
+# Understanding Started.json and Finished.json
+
+### Context
+Prow uploads a host of artifacts throughout the life cycle of a job. Two of these artifacts that are present in each run are `started.json` and `finished.json` which contain a host of information pertaining to the job/run. These files have existed through the evolution of Kubernetes CI: from Jenkins -> Containerized Jenkins -> Bootstrap Containerized Jenkins -> Bootstrap Prow -> PodUtils. As of 2021, all jobs exist within either Bootstrap Prow or PodUtils. As the CI has evolved, so has `started/finished.json` and it's function.
+
+Related Issues:
+
+1. #3412: What is the origin and porous for the fields in these files?
```
Comment (amwat) [inline prow/metadata_artifacts.md:8]: porous?

File: prow/metadata_artifacts.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
+# Understanding Started.json and Finished.json
+
+### Context
+Prow uploads a host of artifacts throughout the life cycle of a job. Two of these artifacts that are present in each run are `started.json` and `finished.json` which contain a host of information pertaining to the job/run. These files have existed through the evolution of Kubernetes CI: from Jenkins -> Containerized Jenkins -> Bootstrap Containerized Jenkins -> Bootstrap Prow -> PodUtils. As of 2021, all jobs exist within either Bootstrap Prow or PodUtils. As the CI has evolved, so has `started/finished.json` and it's function.
+
+Related Issues:
```
Comment (amwat) [inline prow/metadata_artifacts.md:6]: let's also add links to these

**Review by amwat — COMMENTED** [review]
/lgtm
/approve
/hold

minor comments

**Comment by stevekuznetsov** [comment]
Can't say I expected to ever have that issue closed ;)

File: prow/metadata_artifacts.md (hunk 7 above)
Comment (MushuEE) [reply prow/metadata_artifacts.md:6]: Links to examples from a prow run?

File: prow/metadata_artifacts.md (hunk 5 above)
Comment (MushuEE) [reply prow/metadata_artifacts.md:9]: extra `0` thanks

File: prow/metadata_artifacts.md (hunk 7 above)
Comment (MushuEE) [reply prow/metadata_artifacts.md:10]: Added

**Pull request merged**